client-js/app/elements/labrad-plot1d.ts

```diff
@@ -0,0 +1,255 @@
+import 'd3';
+
+var COLOR_LIST = ['red', 'blue', 'yellow', 'green', 'magenta'];
+
+export interface Plot {
+  addData(data: Array<Array<number>>): void
+}
+
+@component('labrad-plot1d')
+export class Plot1D extends polymer.Base implements Plot {
+
+  @property({type: String, value: ''})
+  xLabel: string;
+
+  @property({type: String, value: ''})
+  yLabel: string;
+
+  private data: Array<Array<number>> = []
+  private lastData: Array<number> = null;
+
+  private numTraces: number = 0;
+  private svg: any;
+  private chartBody: any;
+  private clip: any;
+  private xAxis: any;
+  private yAxis: any;
+  private xScale: any;
+  private yScale: any;
+  private line: any;
+  private zoom: any;
+  private limits = {xMin: 0, xMax: 1, yMin: 0, yMax: 1};
+  private dataLimits = {xMin: NaN, xMax: NaN, yMin: NaN, yMax: NaN};
```

Comment:
We need a sentinel to indicate that a limit has not been computed (because we have no data yet). Could alternately use `null`, but `NaN` is an in-band sentinel for numbers, so I went with that.